airflow/contrib/hooks/sqoop_hook.py

```diff
@@ -159,11 +158,12 @@ def _get_export_format_argument(self, file_type='text'):
                                    "'sequence', 'parquet' or 'text'.")
 
     def _import_cmd(self, target_dir, append, file_type, split_by, direct,
-                    driver):
+                    driver, extra_import_options):
 
         cmd = self._prepare_command(export=False)
 
-        cmd += ["--target-dir", target_dir]
+        if target_dir:
```

Comment:
Target directory should be optional because for instance if you try Sqoop with a query, it requires it but if it's with a table you don't have to pass it in.

Reply:
added test to verify if `target_dir` is `None` it doesn't get added

Reply:
👍